phoenix/app/cad_products_phoenix/lib/cad_products_phoenix_web/services/product.ex

```diff
@@ -4,35 +4,33 @@ defmodule CadProductsPhoenixWeb.Services.Product do
   alias CadProductsPhoenix.Management
   alias CadProductsPhoenix.ProductIndex
 
-  def fetch_all() do
-   Cache.get("products")
-    case Cache.get("products") do
-      {:ok, _} = result ->
-        result
-      _ ->
-        register = Management.list_register()
-        Cache.set("products", register)
-        {:ok, register}
-      end
+  def fetch_products(params) do
+    ProductIndex.search_products(params)
   end
 
   def create(%{"product" => product}) when is_map(product) do
     case Management.create_register(product) do
-      {:ok, _} = result ->
-        Cache.delete("products")
-        ProductIndex.index_product(result)
-        result
+      {:ok, product} ->
+        Cache.set(product.id, product)
+        ProductIndex.create_product(product)
+        product
       error -> error
     end
   end
 
   def create(_params), do: {:error, %{code: 422, message: "Unable to create the product, check if you are passing the json correctly"}}
 
+  def show(%{"product" => product}) do
+    product
+  end
+
+  def show(_product), do: {:error, %{code: 422, message: "Unable to find the product, check if you are passing the id correctly"}}
```

Comment:
Agora, como no controller voce esta passando:

```
  def show(conn, _) do
    %{"product" => conn.assigns[:register]}
    |> Product.show()
  end
```
Esse matching nunca sera utilizado. Melhor voltar o controller ao normal e apagar essa função, pois atualmente ambos os show não tem feito nada. Mas eu entendi qual era a intenção e só vai funcionar se esse show for para o controller